src/pages/applicant/viewingKey.tsx

```diff
@@ -164,30 +163,33 @@ const ViewingKeyPage = () => {
           </p>
         </div>
       </div>
-      <div className="w-full text-center z-50 sm:px-20 lg:px-40 flex flex-col ">
-        <form className="flex flex-col items-start mt-8  w-full">
+      <div className="w-full text-center z-50 sm:px-20 lg:px-40 flex flex-col">
+        <form
+          className="flex flex-col items-start mt-8  w-full"
+          onSubmit={handleCreateViewingKey}
+        >
           <label className="text-left mb-1">Viewing key name or phrase</label>
           {inputDataInput((e) => setinputData(e.target.value))}
-        </form>
-        <div className="flex justify-between w-full">
-          <div className="flex items-center mt-8">
-            <Button
-              onClick={() => {
-                router.push(`/applicant`);
-              }}
-              text="Back"
-              style={BUTTON_STYLES.OUTLINE}
-            />
-          </div>
-          <div className="flex items-center mt-8">
-            <Button
-              text={'Create'}
-              classes={{ container: 'mr-3' }}
-              isDisabled={!inputData}
-              onClick={() => handleCreateViewingKey()}
-            />
+          <div className="flex justify-between w-full">
+            <div className="flex items-center mt-8">
+              <Button
+                onClick={() => {
+                  router.push(`/applicant`);
+                }}
+                text="Back"
+                style={BUTTON_STYLES.OUTLINE}
+              />
+            </div>
+            <div className="flex items-center mt-8">
+              <Button
+                text={'Create'}
+                classes={{ container: 'mr-3' }}
+                isDisabled={!inputData}
+                type={BUTTON_ACTION.SUBMIT}
+              />
+            </div>
```

Comment:
I think the back button is triggered here... Also add the check for letters not just spaces before enabling the btn